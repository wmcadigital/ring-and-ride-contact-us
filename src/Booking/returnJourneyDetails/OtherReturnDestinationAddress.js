import { useFormState } from "react-final-form";
import PropTypes from "prop-types";

import FormSection from "../../common/FormSection";
import ProgressIndicator from "../../common/ProgressIndicator";
import Question from "../../common/Question";
import Address from "../../common/Address/Address";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutReturnJourneySectionName from "./getAboutReturnJourneySectionName";

const OtherReturnDestinationAddress = ({
  otherReturnDestinationAddresses,
  setOtherReturnDestinationAddresses,
}) => {
  const stateApi = useFormState();
  const formValues = stateApi.values;
  const bookingParty = formValues["bookingParty"];

  let question = "";
  if (bookingParty === "mySelf") {
    question = "What address do you want to go to?";
  } else if (bookingParty === "behalfSomeone") {
    question = `What address does ${formValues["firstName"]} want to go to?`;
  } else if (bookingParty === "behalfGroup") {
    question = "What address is the group going to?";
  }

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("3", stateApi)}
        sectionName={getAboutReturnJourneySectionName(stateApi)}
      />
      <Question text={question} />
      <Address
        prefix="otherReturnDestination"
        checkInside={true}
        addresses={otherReturnDestinationAddresses}
        setAddresses={setOtherReturnDestinationAddresses}
      />
    </FormSection>
  );
};

OtherReturnDestinationAddress.propTypes = {
  otherReturnDestinationAddresses: PropTypes.arrayOf(PropTypes.object),
  setOtherReturnDestinationAddresses: PropTypes.func,
};

export default OtherReturnDestinationAddress;
