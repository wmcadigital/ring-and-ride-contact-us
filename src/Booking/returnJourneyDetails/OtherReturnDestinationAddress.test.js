import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import OtherReturnDestinationAddress from "./OtherReturnDestinationAddress";

jest.mock("react-final-form");

describe("OtherReturnDestinationAddress", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<OtherReturnDestinationAddress />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfSomeone",
        firstName: "John",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<OtherReturnDestinationAddress />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for a group", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfGroup",
        firstName: "John",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<OtherReturnDestinationAddress />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
