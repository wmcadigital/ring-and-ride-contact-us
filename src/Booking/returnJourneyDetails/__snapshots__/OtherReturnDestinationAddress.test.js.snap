// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherReturnDestinationAddress individual is filling out form for a group 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their return journey"
    sectionPosition="Section 4 of 5"
  />
  <Question
    text="What address is the group going to?"
  />
  <Address
    addresses={Array []}
    checkInside={true}
    prefix="otherReturnDestination"
  />
</FormSection>
`;

exports[`OtherReturnDestinationAddress individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their return journey"
    sectionPosition="Section 3 of 4"
  />
  <Question
    text="What address does John want to go to?"
  />
  <Address
    addresses={Array []}
    checkInside={true}
    prefix="otherReturnDestination"
  />
</FormSection>
`;

exports[`OtherReturnDestinationAddress individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your return journey"
    sectionPosition="Section 3 of 4"
  />
  <Question
    text="What address do you want to go to?"
  />
  <Address
    addresses={Array []}
    checkInside={true}
    prefix="otherReturnDestination"
  />
</FormSection>
`;
